docs/streaming-programming-guide.md

```diff
@@ -644,13 +644,39 @@ methods for creating DStreams from files as input sources.
     </div>
     </div>
 
-	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory (files written in nested directories not supported). Note that
-
-     + The files must have the same data format.
-     + The files must be created in the `dataDirectory` by atomically *moving* or *renaming* them into
-     the data directory.
-     + Once moved, the files must not be changed. So if the files are being continuously appended, the new data will not be read.
-
+	Spark Streaming will monitor the directory `dataDirectory` and process any files created in that directory.
+
+     + A simple directory can be supplied (`hdfs://namenode:8040/logs/`). All files directly
+       underneath this path will be processed as they are discovered.
+     + A regular expression can be supplied instead, such as
+       `s3a://bucket/logs/[2015,2016]-??-??-friday`.
```

Comment:
You'll need (2015|2016) rather than (2015,2016). Also this is going to match zero or more hyphens followed by "-friday". I think you mean ".." or ".{2}" or at least ".+" instead of "*" if this is a regex.